### PR TITLE
docs: Update Excel page of user guide to refer to fastexcel as the default engine

### PR DESCRIPTION
**Pull request by braaannigan**
The user guide page for Excel files has not been updated to account for calamine as the default reader. The PR updates the page along with some minor edits

### CODE REVIEW

File: docs/source/user-guide/io/excel.md [hunk 1]
```diff
@@ -5,25 +5,23 @@ From a performance perspective, we recommend using other formats if possible, su
...
+Polars does not have a native Excel reader. Instead, it uses an external library called an "engine" to parse Excel files into a form that Polars can parse. The available engines are:
 
-- xlsx2csv: This is the current default.
+- fastexcel: This reader is based on the Rust [calamine](https://github.com/tafia/calamine) crate and is typically the fastest reader.
+- xlsx2csv: This reader parses the .xlsx file to an in-memory CSV that Polars then reads with its own CSV reader.
 - openpyxl: Typically slower than xls2csv, but can provide more flexibility for files that are difficult to parse.
-- fastexcel: This reader is based on [calamine](https://github.com/tafia/calamine) and is typically the fastest reader but has fewer features than xls2csv.
 
-Although fastexcel is not the default at this point, we recommend trying fastexcel first and using xlsx2csv or openpyxl if you encounter issues.
+We recommend working with the default fastexcel engine. The xlsx2csv and openpyxl engines are slower but do have more features for parsing tricky data. These engines may be helpful if the fastexcel reader does not work for a specific Excel file.
```
Comment (alexander-beedie) [inline docs/source/user-guide/io/excel.md:14]:
```suggestion
We recommend working with the default fastexcel engine. The xlsx2csv and openpyxl engines are slower but may have more features for parsing tricky data. These engines may be helpful if the fastexcel reader does not work for a specific Excel file.
```

Comment (alexander-beedie) [reply docs/source/user-guide/io/excel.md:14]: Let's soften that one a bit; the `fastexcel`  folks have added quite a few features over the last few months and it's getting pretty fully featured :)

**Review by alexander-beedie — CHANGES_REQUESTED** [review]
Good catch; just a minor nit :)

File: docs/source/user-guide/io/excel.md [hunk 2]
```diff
@@ -5,25 +5,23 @@ From a performance perspective, we recommend using other formats if possible, su
 
 ## Read
 
-Polars does not have a native Excel reader. Instead, it uses external libraries to parse Excel files into objects that Polars can parse. The available engines are:
+Polars does not have a native Excel reader. Instead, it uses an external library called an "engine" to parse Excel files into a form that Polars can parse. The available engines are:
 
-- xlsx2csv: This is the current default.
+- fastexcel: This reader is based on the Rust [calamine](https://github.com/tafia/calamine) crate and is typically the fastest reader.
```
Comment (alexander-beedie) [inline docs/source/user-guide/io/excel.md:10]:
```suggestion
- fastexcel: This engine is based on the Rust [calamine](https://github.com/tafia/calamine) crate and is (by far) the fastest reader.
```

**Review by alexander-beedie — APPROVED** [review]

**Comment by braaannigan** [comment]
@alexander-beedie am I right that you don't need anything further from me on this? Not sure why that test is failing, the rust analyser link works when I try it.

**Comment by alexander-beedie** [comment]
> @alexander-beedie am I right that you don't need anything further from me on this? Not sure why that test is failing, the rust analyser link works when I try it.

Yup, I think it's good to merge - also not sure why that error appeared :\

**Pull request merged**